crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -2027,6 +2080,28 @@ impl<C: Comments> VisitMut for ServerActions<C> {
         self.arrow_or_fn_expr_ident = old_arrow_or_fn_expr_ident;
     }
 
+    fn visit_mut_this_expr(&mut self, n: &mut ThisExpr) {
+        if let ThisStatus::Forbidden { directive } = &self.this_status {
+            emit_error(ServerActionsErrorKind::ForbiddenExpression {
+                span: n.span,
+                expr: "this".into(),
+                directive: directive.clone(),
+            });
+        }
+    }
+
+    fn visit_mut_ident(&mut self, n: &mut Ident) {
+        if n.sym == *"arguments" {
```

Comment:
i'm actually not sure if this valid, apparently `arguments` is also a valid variable name...
```
> function test() { const arguments = ["nope"]; return arguments }
> test("yep")
Array [ "nope" ]
```
so unless we also track assignments to `arguments`, we can't be sure that this is invalid

Reply:
SWC also won't compile this and error with `'eval' and 'arguments' cannot be used as a binding identifier in strict mode`.